client/tokenized-payment-request/order-api.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { getPaymentRequestData } from './frontend-utils';
+
+export default class PaymentRequestOrderApi {
+	// parameters used in every request, just in different ways.
+	orderId;
+	key;
+	billingEmail = '';
+
+	/**
+	 * Creates an instance of class to query for order data.
+	 *
+	 * @param {string} orderId The order ID,
+	 * @param {string} key The order key, used to verify the order ID.
+	 * @param {string?} billingEmail The billing email address, used for guest orders.
+	 */
+	constructor( { orderId, key, billingEmail = '' } ) {
+		this.orderId = orderId;
+		this.key = key;
+		this.billingEmail = billingEmail;
+	}
+
+	/**
+	 * Creates an order from the cart object.
+	 * See https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/docs/checkout-order.md
+	 *
+	 * @param {{
+	 *          billing_address: Object,
+	 *          shipping_address: Object,
+	 *          payment_method: string,
+	 *          payment_data: Array,
+	 *        }} paymentData Additional payment data to place the order.
+	 * @return {Promise} Result of the order creation request.
+	 */
+	async placeOrder( paymentData ) {
+		return await apiFetch( {
+			method: 'POST',
+			path: `/wc/store/v1/checkout/${ this.orderId }`,
+			headers: {
+				Nonce: getPaymentRequestData( 'nonce' ).tokenized_order_nonce,
+			},
+			data: {
+				...paymentData,
+				key: this.key,
+				billing_email: this.billingEmail,
+				// preventing billing and shipping address from being overwritten in the request to the store - we don't want to update them
+				billing_address: undefined,
+				shipping_address: undefined,
+			},
+		} );
+	}
+
+	/**
+	 * Returns the customer's order object.
+	 * See https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/docs/cart.md#get-cart
+	 *
+	 * @return {Promise} Cart response object.
+	 */
+	async getCart() {
```

Comment:
I initially called this method `getOrder`, but then I thought of renaming it `getCart`. After all, it's returning a cart response object. And renaming it slightly simplifies the branching logic 6e4d8980b